### ISSUE TITLE
qgis-26 links to nonexistent libexpat

### ISSUE THREAD

**Issue by jconley** [OSGeo/homebrew-osgeo4mac#70]
qgis-26 builds fine on Yosemite 10.10.1 Xcode: 6.1.1, but upon launching it tries to reference libexpat in the wrong location, and seems to fall back to the system version which does not meet the minimum version requirement.

```
$ qgis
dyld: Library not loaded: /usr/local/lib/libexpat.1.dylib
  Referenced from: /usr/local/opt/qgis-26/QGIS.app/Contents/MacOS/QGIS
  Reason: Incompatible library version: QGIS requires version 8.0.0 or later, but libexpat.1.dylib provides version 7.0.0
/usr/local/bin/qgis: line 9: 77383 Trace/BPT trap: 5       /usr/local/opt/qgis-26/QGIS.app/Contents/MacOS/QGIS "$@"
```

libexpat is keg-only and only resides in /usr/local/opt/expat/lib/...

```
$ otool -L /usr/local/opt/qgis-26/QGIS.app/Contents/MacOS/QGIS
/usr/local/opt/qgis-26/QGIS.app/Contents/MacOS/QGIS:
...
	/usr/local/lib/libexpat.1.dylib (compatibility version 8.0.0, current version 8.0.0)
	/usr/local/opt/sqlite/lib/libsqlite3.0.dylib (compatibility version 9.0.0, current version 9.6.0)
```

`ls: /usr/local/lib/libexpat.1.dylib: No such file or directory`

manually changing the shared lib paths resolves the issue:
`install_name_tool -change /usr/local/lib/libexpat.1.dylib /usr/local/opt/expat/lib/libexpat.1.dylib`

against these 4 files:
```
/usr/local/opt/qgis-26/QGIS.app/Contents/MacOS/QGIS
/usr/local/Cellar/qgis-26/2.6.1/QGIS.app/Contents/Frameworks/qgis_gui.framework/Versions/2.6/qgis_gui
/usr/local/Cellar/qgis-26/2.6.1/QGIS.app/Contents/Frameworks/qgis_core.framework/Versions/2.6/qgis_core
/usr/local/Cellar/qgis-26/2.6.1/QGIS.app/Contents/Frameworks/qgis_analysis.framework/Versions/2.6/qgis_analysis
```

need to get the build to properly link to the keg-opt location

**Comment by dakcarto** [comment]
Hi @jconley. Thanks for reporting.

I think you are running into the same `expat` issue [previously reported](https://github.com/OSGeo/homebrew-osgeo4mac/issues/45#issuecomment-54165311), whereby the `expat` formula was moved to keg-only, but the bottle still had the incorrect install name (with the `/usr/local/lib` path).

Try:

```
brew update
brew reinstall expat
brew reinstall qgis-26
```
The reinstall of `qgis-26` is needed to properly link to the updated `expat` install.

Basically, the `expat` formula was moved to keg-only, but was not given a `revision`, which means anyone with the older install, with the screwed-up install name, was never prompted to upgrade. Its formula needs that revision bump.

**Comment by jconley** [comment]
that fixed it, thanks
